presto-spark-base/src/main/java/com/facebook/presto/spark/execution/nativeprocess/HttpNativeExecutionTaskResultFetcher.java

```diff
@@ -136,101 +145,90 @@ public Optional<SerializedPage> pollPage()
 
     public boolean hasPage()
     {
+        if (scheduledFuture != null && scheduledFuture.isCancelled() && lastException.get() != null) {
+            throw lastException.get();
+        }
+
         return !pageBuffer.isEmpty();
     }
 
-    private static class HttpNativeExecutionTaskResultFetcherRunner
-            implements Runnable
+    private void doGetResults()
+    {
+        if (bufferMemoryBytes.longValue() >= MAX_BUFFER_SIZE.toBytes()) {
+            return;
+        }
+
+        try {
+            PageBufferClient.PagesResponse pagesResponse = workerClient.getResults(token, MAX_RESPONSE_SIZE).get();
+            onSuccess(pagesResponse);
+        }
+        catch (InterruptedException | ExecutionException e) {
```

Comment:
Should we expand the net here to all exceptions ? Because any uncaught exception would result in halting the subsequent executions of `scheduleAtFixedRate`. We only want the `stop` function to be one to halt subsequent executions, instead of it happening here silently due to uncaught exception